config/management/src/smoke_test.rs

```diff
@@ -98,79 +110,219 @@ fn smoke_test() {
         }
         config.execution.genesis = Some(genesis.clone());
         config.execution.genesis_file_location = PathBuf::from("");
     }
 
-    // Step 6) Prepare ecosystem
-    let full_node_config = attach_validator_full_node(&mut configs[0]);
-
-    // Step 7) Build configuration for Swarm
-    let faucet_key = helper
-        .storage(association.into())
-        .export_private_key(libra_global_constants::ASSOCIATION_KEY)
-        .unwrap();
-    let management_builder = ManagementBuilder {
-        configs,
-        faucet_key: faucet_key.clone(),
-    };
-
-    let mut swarm = LibraSwarm {
-        dir: LibraSwarmDir::Temporary(temppath),
-        nodes: std::collections::HashMap::new(),
-        config: SwarmConfig::build(&management_builder, &swarm_path).unwrap(),
-    };
-
-    // Step 8) Launch validators
-    swarm.launch_attempt(RoleType::Validator, false).unwrap();
-
-    // Step 9) Launch ecosystem
-    let management_builder = ManagementBuilder {
-        configs: vec![full_node_config],
-        faucet_key,
-    };
+    fn create_swarm(self) -> LibraSwarm {
+        let config = SwarmConfig::build(&self, &self.swarm_path.path().to_path_buf()).unwrap();
+        LibraSwarm {
+            dir: LibraSwarmDir::Temporary(self.swarm_path),
+            nodes: std::collections::HashMap::new(),
+            config,
+        }
+    }
+}
 
-    let temppath = TempPath::new();
-    temppath.create_as_dir().unwrap();
-    let swarm_path = temppath.path().to_path_buf();
+impl BuildSwarm for ManagementBuilder {
+    fn build_swarm(&self) -> anyhow::Result<(Vec<NodeConfig>, Ed25519PrivateKey)> {
+        self.create_layout();
+        self.create_association();
+        let faucet_key = self
+            .storage_helper
+            .storage(ASSOCIATION.into())
+            .export_private_key(libra_global_constants::ASSOCIATION_KEY)
+            .unwrap();
+        let mut configs: Vec<_> = (0..self.num_validators)
+            .map(|i| self.initialize_validator_config(i))
+            .collect();
+        let waypoint = self
+            .storage_helper
+            .create_waypoint(constants::COMMON_NS)
+            .unwrap();
+        configs
+            .iter_mut()
+            .enumerate()
+            .for_each(|(i, config)| self.finish_validator_config(i, config, waypoint));
+        Ok((configs, faucet_key))
+    }
+}
 
-    let mut swarm = LibraSwarm {
-        dir: LibraSwarmDir::Temporary(temppath),
-        nodes: std::collections::HashMap::new(),
-        config: SwarmConfig::build(&management_builder, &swarm_path).unwrap(),
-    };
+#[test]
+fn smoke_test() {
+    LibraNode::prepare();
+    let mut swarm = ManagementBuilder::new(5).create_swarm();
+    swarm.launch_attempt(RoleType::Validator, false).unwrap();
 
-    swarm.launch_attempt(RoleType::FullNode, false).unwrap();
-    assert!(check_connectivity(&mut swarm, 1));
+    // let helper = StorageHelper::new();
```

Comment:
nit: is there any benefit in leaving this here if it just gets removed in the next commit? Might be cleaner to just delete this now, no?

Reply:
oh no, it's just for me to easily copy the code around lol